Go post-1.5.2 has broken the proxy used in api/lock_test.go

Something in the latest tip Go build has broken the proxy used in the lock monitor test. Here's a build when it started happening:

https://s3.amazonaws.com/archive.travis-ci.org/jobs/97105387/log.txt

```
--- FAIL: TestLock_MonitorRetry (2.48s)
	lock_test.go:447: err: failed to create session: Unexpected response code: 500 ()
```

Adding some debugging prints shows that the session create request is successfully going to the "raw" connection via the proxy, but it's giving a 500 back with an empty response. This is suspiciously like what the "failer" server does, but it's not clear it's hitting that.

We will pin the Travis builds at 1.5.2 for now to get it going again but when we get a few minutes we should dig on the Go side and see if we can help identify what happened.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: go
 
 go:
-  - 1.5.2
+  - tip
 
 branches:
   only:

